[tools/shoestring]: Shortened command and subcommand autocompletion

ccHarvestasya opened the pull request:
## Problem

- Running the tool requires prefixing with `python3 -m`
- No autocomplete support for subcommands

## Solution

- Define Poetry scripts to eliminate the need for `python3 -m`
- Enable subcommand autocompletion using `argcomplete`

> This is not something I use frequently, but it’s a nice-to-have improvement.

ccHarvestasya committed:
[tools/shoestring] feat: add Poetry scripts for shoestring and wizard entry points

ccHarvestasya committed:
[tools/shoestring] feat: add autocomplete support via argcomplete

Enable shell autocompletion using:
  eval "$(register-python-argcomplete shoestring)"

ccHarvestasya committed:
[tools/shoestring] fix: linting error

Wayonb commented on tools/shoestring/pyproject.toml:
```diff
@@ -23,5 +23,9 @@ classifiers = [
 
+[tool.poetry.scripts]
+shoestring = "shoestring.__main__:entry_point"
+shoestring-wizard = "shoestring.wizard.__main__:entry_point"
```
Can you add tests for this?

ccHarvestasya replied:
It is possible to run tests that go through each entry point, so I’ve added them.
However, these are just tests that pass through the entry points.
Do you think it would be better to include some assertions?

Wayonb replied:
Thanks. Those tests are fine.
Initially, I was thinking of installing a Shoestring(pip install -e .) to verify that the ``shoestring`` command runs.

ccHarvestasya committed:
[tools/shoestring] fix: add test for entry_point

ccHarvestasya committed:
[tools/shoestring] fix: entry_point test

ccHarvestasya committed:
[tools/shoestring] fix: entry_point test more

Wayonb commented on tools/shoestring/requirements.txt:
```diff
@@ -1,4 +1,5 @@
 aiohttp~=3.12.6
+argcomplete~=3.6.2
```
1. Yes, but I was wondering if ``argcomplete`` only supports bash or zsh shell

2. Yes, will need to update the README on how to enable autocompletion

ccHarvestasya replied:
1. It looks like `argcomplete` only supports Bash and Zsh.  
Are there any other shells we should consider supporting? For example, PowerShell?  
If you have suggestions for other shells worth supporting, let me know — I’ll look into whether we can support them.

1. I'll add instructions for enabling autocomplete to the README later.

Wayonb replied:
1. Adding the list of supported shells to the README is sufficient.

ccHarvestasya committed:
[tools/shoestring] add autocomplete activation

ccHarvestasya committed:
Merge branch 'dev' into shoestring/add_entrypoints_to_poetry

AnthonyLaw committed:
[lightapi/python] task: add NemConnector endpoints

task: add NemConnector for POST local/chain/blocks-after and local/block/at
task: add Block model
task: add mapping transaction, TransferTransaction, ImportanceTransferTransaction, and ConvertAccountToMultisig
task: refactor the transaction model
task: add new model NamespaceRegistrationTransaction, MosaicDefinitionTransaction, and MosaicSupplyChangeTransaction
task: refactor and support more transactions
task: add new model MultisigTransaction and CosignSignatureTransaction
task: refactor the transaction model and the NEM connector
task: rename transaction model
task: add UnknownTransactionType exception

---
    
Signed-off-by: wayonb <wayon@symbol.dev>
Co-authored-by: Anthony Law <anthony@symbol.dev>

ccHarvestasya committed:
Merge branch 'dev' into shoestring/add_entrypoints_to_poetry
